xtool/xtool/version/manager.py

```diff
@@ -86,6 +87,18 @@ def migrateVersions(self):
     def hasVersion(self, version):
         return version in self.configManager.versions()
 
+    def _normalizedVersion(self, version):
+        # Convince packaging.version to like our Java versions
+        # by replacing SNAPSHOT with a, and some dashes to dots.
+        return Version.parse(re.sub(r"([0-9])-([0-9])", r"\1.\2", version.replace("SNAPSHOT", "a")))
```

Comment:
Does it work with the RC versions?

Reply:
Sadly,

```
>>> _normalizedVersion("10.8-rc-1") > _normalizedVersion("10.8-SNAPSHOT")
True
```

Suggestions welcome

Reply:
I'll try mapping "rc" to "a" (alpha) and SNAPSHOT to "b" (beta)

Reply:
> I think I would only allow upgrading to a snapshot with a force flag if you're willing to only do it by version matching

Somehow your message didn't show up before my last one: I like the idea of introducing a force flag. The force flag would skip the version matching altogether. We would prevent upgrading to SNAPSHOT by default and not do any version matching in case the force flag is used. This would allow to get rid of this fragile regex replacing.
WDYT @surli?

Reply:
Sounds good to me yes.

Reply:
Done at https://github.com/xwiki-contrib/xtool/pull/3, closing this one.